src-tauri/src/main.rs

```diff
@@ -94,582 +105,483 @@ fn find_configuration_endpoints<T: UsbContext>(
+                    });
+                }
             }
         }
     }
     None
 }
 
-#[repr(u16)]
-#[allow(dead_code)]
-enum StructureTypes {
-    // Commands/Responses, these are container TLVs. The Value will be a set of TLV structures.
-    OK = 0,                     // Standard response when a command was successful
-    NOK,                        // Standard error response
-    FlashHeader,                // A special container for the config stored in flash. Hopefully there is some useful
-                                // metadata in here to allow us to migrate an old config to a new version.
-    GetVersion,                 // Returns the current config version, and the minimum supported version so clients
-                                // can decide if they can talk to us or not.
-    SetConfiguration,           // Updates the active configuration with the supplied TLVs
-    GetActiveConfiguration,     // Retrieves the current active configuration TLVs from RAM
-    GetStoredConfiguration,     // Retrieves the current stored configuration TLVs from Flash
-    SaveConfiguration,          // Writes the active configuration to Flash
-    FactoryReset,               // Invalidates the flash memory
-
-    // Configuration structures, these are returned in the body of a command/response
-    PreProcessingConfiguration = 0x200,
-    FilterConfiguration,
-    Pcm3060Configuration,
-
-    // Status structures, these are returned in the body of a command/response but they are
-    // not persisted as part of the configuration
-    VersionStatus = 0x400,
-}
-
-enum FilterType {
-    Lowpass = 0,
-    Highpass,
-    BandpassSkirt,
-    BandpassPeak,
-    Notch,
-    Allpass,
-    Peaking,
-    LowShelf,
-    HighShelf,
-    CustomIIR
+#[derive(Serialize, Deserialize, Default, Debug)]
+pub struct Preprocessing {
+    preamp: f32,
+    post_eq_gain: f32,
+    reverse_stereo: bool,
 }
 
-#[derive(Serialize, Deserialize, Default)]
-struct Filter {
-    filter_type: String,
-    q: f32,
-    f0: f32,
-    db_gain: f32,
-    a0: f64,
-    a1: f64,
-    a2: f64,
-    b0: f64,
-    b1: f64,
-    b2: f64,
-    enabled: bool
-}
+impl Preprocessing {
+    fn new(preamp: f32, post_eq_gain: f32, reverse_stereo: bool) -> Self {
+        Preprocessing {
+            preamp: preamp.log10() * 20.0,
+            post_eq_gain: post_eq_gain.log10() * 20.0,
+            reverse_stereo,
+        }
+    }
 
-#[derive(Serialize, Deserialize, Default)]
-struct Preprocessing {
-    preamp: f32,
-    postEQGain: f32,
-    reverse_stereo: bool
+    fn to_payload(&self) -> Vec<u8> {
+        let mut preprocessing_payload: Vec<u8> = Vec::new();
+        // TODO: -1.0 as the firmware adds 1, cleanup later. Consider storing this value without the subtraction
+        // to eliminate a math op and make the code more grokable?
+        //preprocessing_payload.extend_from_slice(&(f32::powf(10.0, cfg.preprocessing.preamp/20.0) - 1.0).to_le_bytes());
```

Comment:
I guess we can delete this commented out line?

Reply:
Yeah, looks not important, let's delete it